### PR TITLE
[Keyboard]Add X-Bows Ranger Keyboard

### PR DESCRIPTION
**Pull request by XBowsTech**
## Description

Add support for X-Bows Ranger Keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

File: keyboards/xbows/ranger/ranger.c [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+ #include "ranger.h"
+ #include "rgblight.h"
```
Comment (fauxpark) [inline keyboards/xbows/ranger/ranger.c:17]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

File: keyboards/xbows/ranger/readme.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Ranger
+![Ranger](https://img.alicdn.com/i3/16731204/O1CN015gRK1R1KlTSHEuHDo_!!16731204.jpg)
+
+Ranger, A 87 keys keyboard with RGB backlight.
+
+* Keyboard Maintainer: X-BOWS
+* Hardware Supported: X-BOWS Ranger
+* Hardware Availability: [X-BOWS](https://x-bows.com/)
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/xbows/ranger/readme.md:10]: Could you add instructions on how to enter the bootloader, to flash new firmware?

Comment (XBowsTech) [reply keyboards/xbows/ranger/readme.md:10]: Enter the bootloader in 3 ways:
* Software reset on Fn + ESC
* Bootmagic reset: hold down the top left key (usually escape) and plugin the keyboard
* Physical reset button: There is a button on the back of the PCB, please press it briefly.

Flash new firmware in 2 ways:
* Use the QMK toolbox：https://github.com/qmk/qmk_toolbox/releases
* Use the QMK MSYS：https://msys.qmk.fm/

Comment (XBowsTech) [reply keyboards/xbows/ranger/readme.md:19]: > Could you add instructions on how to enter the bootloader, to flash new firmware?

I added some instructions. Do I need to make any other changes?

**Comment by drashna** [comment]
Thanks!

**Comment by drashna** [comment]
For future reference, we recommend against committing to your `master` branch as you've done here, because pull requests from modified `master` branches can make it more difficult to keep your QMK fork updated. It is highly recommended for QMK development – regardless of what is being done or where – to keep your master updated, but **NEVER** commit to it. Instead, do all your changes in a branch (branches are basically free in Git) and issue PRs from your branches when you're developing.

There are instructions on how to keep your fork updated here:

[**Best Practices: Your Fork's Master: Update Often, Commit Never**](https://docs.qmk.fm/#/newbs_git_using_your_master_branch)

[Fixing Your Branch](https://docs.qmk.fm/#/newbs_git_resynchronize_a_branch) will walk you through fixing up your `master` branch moving forward. If you need any help with this just ask.

**You do not need to close this pull request at this time.** Until your pull request is merged, please continue to commit relevant changes to your `master` branch.

**Pull request merged**